src/sql/query.rs

```diff
@@ -1707,17 +1707,26 @@ fn plan_function<'a>(
                 }
 
                 let source_timestamp =
-                    plan_expr(ecx, &sql_func.args[1], Some(ScalarType::Timestamp))?;
+                    plan_expr(ecx, &sql_func.args[1], Some(ScalarType::TimestampTz))?;
```

Comment:
I believe the type hint here should actually be `None`, since we don't know whether to pick timestamp or timestamptz:

```
benesch=# prepare f as select date_trunc('day'::text, $1);
ERROR:  function date_trunc(text, unknown) is not unique
LINE 1: prepare f as select date_trunc('day'::text, $1);
```

Reply:
This is so minor, too, that there's no need to go back through CI for it. It will eventually be fixed by #1381 anyway.